File: online-admission-backend/src/main/java/com/cg/admission/service/UniversityService.java
package com.cg.admission.service;

import java.util.List;

import com.cg.admission.dto.CollegeDto;
import com.cg.admission.dto.UniversityDto;
import com.cg.admission.entity.University;

public interface UniversityService {

	List<UniversityDto> getAllUniversity();

	UniversityDto getUniversityById(Long Id);

	UniversityDto addUniversity(University university);

	List<UniversityDto> getUniversityDetailsByCity(String city);

	List<UniversityDto> getUniversityDetailsByCollegeName(String collegeName);

	Long updateUniversity(Long Id, University university);

	String deleteUniversityById(Long id);

	List<CollegeDto> findCollegeByUniverisity(Long universityId);
}

File: online-admission-system-frontend/online-admission-system-ui/src/Services/UniversityService.js
import axios from 'axios';

const REST_API_BASE_URL_UNIVERSITY = 'http://localhost:8080/university';


// export const getAllColleges = (universityId) =>{
//     return axios.get(REST_API_BASE_URL_UNIVERSITY +'/'+ universityId+'/colleges');
// }

export const getUniversityDetailsByCity = (city) =>{
    return axios.get(REST_API_BASE_URL_UNIVERSITY +'/city/'+ city); 
}

export const getAllUniversity = () =>{
    return axios.get(REST_API_BASE_URL_UNIVERSITY);
}

export const getUniversityById = (Id) =>{
    return axios.get(REST_API_BASE_URL_UNIVERSITY+'/id/'+Id);
}

export const addUniversity = (universityObj) =>{
    return axios.post(REST_API_BASE_URL_UNIVERSITY,universityObj);
}

export const deleteUniversityById = (Id) =>{
    return axios.delete(REST_API_BASE_URL_UNIVERSITY+'/id/'+Id);
}

export const updateUniversity =(Id, universityObj) =>{
    return axios.put(REST_API_BASE_URL_UNIVERSITY+'/update/'+Id, universityObj);    
}

export const findCollegeByUniverisityId = (universityId) =>{
    return axios.get(REST_API_BASE_URL_UNIVERSITY +'/'+universityId+'/colleges');
}

